src/Quartz/Impl/Calendar/DailyCalendar.cs

```diff
@@ -479,7 +481,7 @@ public override bool IsTimeIncluded(DateTimeOffset timeUtc)
         /// <seealso cref="ICalendar.GetNextIncludedTimeUtc"/>
         public override DateTimeOffset GetNextIncludedTimeUtc(DateTimeOffset timeUtc)
         {
-            DateTimeOffset nextIncludedTime = timeUtc.AddMilliseconds(OneMillis);
+            DateTimeOffset nextIncludedTime = timeUtc.AddMilliseconds(OneSecondInMillis);
```

Comment:
Can we determine the used precision based on configuration? Like if rangeStarting/Ending millis and seconds are zero we can safely search with seconds precision? For v4 we could probably just drop the whole millisecond precision support, if not even seconds support as they both sound a bit too fine-grained for daily operation.

Reply:
Good point, I've amended to code to adjust stepping based on start/end range.